arch/arm/src/arm/arm_syscall.c

```diff
@@ -54,19 +54,23 @@
 
 uint32_t *arm_syscall(uint32_t *regs)
 {
-  struct tcb_s *tcb = this_task();
+  struct tcb_s **running_task = &g_running_tasks[this_cpu()];
+  FAR struct tcb_s *tcb = this_task();
   uint32_t cmd;
-  int cpu;
 
   /* Nested interrupts are not supported */
 
   DEBUGASSERT(up_current_regs() == NULL);
 
+  if (*running_task != NULL)
+    {
+      (*running_task)->xcp.regs = regs;
```

Comment:
What's wrong with simply; 

`tcb->xcp.regs = regs ` ? 

Why do you need to involve running_tasks ?

Reply:
In many cases tcb->xcp.regs is the regs that we are ready to use, It shouldn't be changed to an incorrect value.
The reason why there is no error currently is because *(uint32_t **)regs[REG_R1] = regs; reassigns tcb->xcp.regs to the correct value. Therefore, we made this change

Reply:
So this is a result of how we do the context switch via invoking a trap; the task that is being switched is in "running_task" and the task that will run next is in "this_task()" ?

Reply:
yes, 
Another reason why there isn't a problem currently is because the parameter we pass through the system call is &this_task()->xcp.regs. However, in some architectures, such as RISC-V, the parameter passed is this_task(). In such cases, once we modify this_task()->xcp.regs to an incorrect value, we cannot restore it to the correct value.

In the future, we hope to align the parameters passed by the system call with RISC-V and pass the  TCB (Task Control Block), for example:

riscv_switchcontext(rtcb, tcb);

Reply:
Understood. 

I agree with changing the parameters for xxx_switchcontext to (rtcb, tcb), which is exactly why I changed the parameters in RISC-V. It cannot be assumed that the context switch routine only needs the integer register set.